### PR TITLE
sway-unwrapped: 1.9 -> 1.10

### PR DESCRIPTION
**Pull request by Scrumplex**
https://github.com/swaywm/sway/releases/tag/1.10

While the build option for XWayland was dropped `xcbutilwm` is still required.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by teutat3s — COMMENTED** [review]
Should we include `grim`, `brightnessctl` + `pulseaudio` in the `default` of `programs.sway.extraPackages`? They are now part of the default sway config file.

**Comment by MithicSpirit** [comment]
Result of `nixpkgs-review pr 351752` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>nwg-panel</li>
    <li>nwg-panel.dist</li>
    <li>sway</li>
    <li>sway-contrib.grimshot</li>
    <li>sway-contrib.grimshot.man</li>
    <li>sway-unwrapped</li>
    <li>waybar</li>
    <li>wlprop</li>
  </ul>
</details>

**Review by MithicSpirit — APPROVED** [review]
Works for me. Tested with this overlay:
```nix
  sway-unwrapped =
    (prev.sway-unwrapped.override { wlroots = final.wlroots_0_18; }).overrideAttrs
      (
        finalAttrs: prevAttrs: {
          version = "1.10";
          src = prevAttrs.src.override {
            hash = "sha256-PzeU/niUdqI6sf2TCG19G2vNgAZJE5JCyoTwtO9uFTk=";
          };

          mesonFlags =
            let
              inherit (final.lib.strings) mesonEnable mesonOption;
              sd-bus-provider = if finalAttrs.systemdSupport then "libsystemd" else "basu";
            in
            [
              (mesonOption "sd-bus-provider" sd-bus-provider)
              (mesonEnable "tray" finalAttrs.trayEnabled)
            ];
        }
      );
```

**Comment by Scrumplex** [comment]
I have removed swayidle and swaylock, as those are not used in the default config, but only mentioned as options. I can add them back if someone thinks it might break someone's config

**Comment by teutat3s** [comment]
`grim` is still missing in extraPackages.

File: nixos/modules/programs/wayland/sway.nix [hunk 1]
```diff
@@ -85,9 +85,10 @@ in
 
     extraPackages = lib.mkOption {
       type = with lib.types; listOf package;
-      default = with pkgs; [ swaylock swayidle foot dmenu wmenu ];
+      # Packages used in default config
+      default = with pkgs; [ brightnessctl foot grim pactl wmenu ];
       defaultText = lib.literalExpression ''
-        with pkgs; [ swaylock swayidle foot dmenu wmenu ];
+        with pkgs; [ brightnessctl foot grim pactl wmenu ];
```
Comment (teutat3s) [inline nixos/modules/programs/wayland/sway.nix:91]:
```suggestion
        with pkgs; [ brightnessctl foot grim pulseaudio wmenu ];
```

**Review by teutat3s — COMMENTED** [review]
Thank you for adding these. I cannot find the package `pactl` in nixpkgs, the binary is only part of `pulseaudio` AFAIK.

**Comment by teutat3s** [comment]
> I have removed swayidle and swaylock, as those are not used in the default config, but only mentioned as options. I can add them back if someone thinks it might break someone's config

I wouldn't remove those, please revert that change.

**Review by teutat3s — APPROVED** [review]
Thanks!

**Comment by Scrumplex** [comment]
Currently running this branch on my system and everything seems to work!

**Review by sg-qwt — APPROVED** [review]
Doesn't seem maintainers are very active, can someone merge this?

**Pull request merged**